src/primitives/line/styled.rs

```diff
@@ -64,7 +160,61 @@ impl<C: PixelColor> StyledDrawable<PrimitiveStyle<C>> for Line {
     where
         D: DrawTarget<Color = C>,
     {
-        target.draw_iter(StyledPixelsIterator::new(self, style))
+        let dot_size = style.stroke_width as i32;
+        if dot_size == 0 {
```

Comment:
The cast from `u32` to `i32` could theoretically generate a negative value for `dot_size`, so this would be a bit safer:
```suggestion
        if dot_size <= 0 {
```